### PR TITLE
Return 404 if pipeline not exist and handle 404 errors in WebUI

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

File: server/api/pipeline.go [hunk 1]
```diff
@@ -123,6 +123,10 @@ func GetPipeline(c *gin.Context) {
 
 	pl, err := _store.GetPipelineNumber(repo, num)
 	if err != nil {
+		if errors.Is(err, types.RecordNotExist) {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
```
[the remark this answers is not in the document]
Comment (6543) [reply server/api/pipeline.go:129]: 404 error is not bad as it's defined clearly

[the remark this answers is not in the document]
Comment (6543) [reply server/api/pipeline.go:129]: for the WebUI part ... well we can handle that better but for now it's better to show a defined error than just "we got an error... deal with it"

[the remark this answers is not in the document]
Comment (lafriks) [reply server/api/pipeline.go:129]: I agree on 404 being correct result combined with UI handling it correctly to display that information

**Review by lafriks — APPROVED** [review]
Imho it's a good start but can be improved in the future with 404 page

File: server/api/pipeline.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (6543) [reply server/api/pipeline.go:129]: @qwerty287 I agree that there could be a nice 404 page instead ... but that's out of my vue3 knowledge right now :sweat_smile:

[the remark this answers is not in the document]
Comment (6543) [reply server/api/pipeline.go:129]: I'll merge this and open an issue to address this ...

**Pull request merged**